app/controllers/delivery/book_van.js

```diff
@@ -122,7 +123,9 @@ export default addressDetails.extend({
         order.set('removeNetFee', data.breakdown.remove_net && data.breakdown.remove_net.value);
         loadingView.destroy();
         controller.transitionToRoute('delivery.confirm_van', { queryParams: { placeOrder: true } });
-      });
+      }).catch((error)=>{
+        this.get("messageBox").alert(error.responseJSON.errors[0].message);
```

Comment:
Do we need to check presence of error? or we are pretty sure that it will be present always?

Reply:
Yeah.. I am explicitly sending error from API side.

Reply:
I think what @swatijadhav means, is that the error is not necessarily an API error. Imagine `coupon` is undefined, and you receive an error saying `"ReferenceError: coupon.value is not defined"`

So even if the API explicitly sets the error messages, we're not sure that `error` is that.

You can use lodash and do `_.get(error, 'responseJSON.errors[0].message')` which is safe